Source/Core/Core/IOS/Network/MACUtils.cpp

```diff
@@ -30,19 +31,22 @@ void GetMACAddress(u8* mac)
   if (Core::WantsDeterminism())
     wireless_mac = "12:34:56:78:9a:bc";
 
-  if (!Common::StringToMacAddress(wireless_mac, mac))
+  std::optional<Common::MACAddress> mac = Common::StringToMacAddress(wireless_mac);
+
+  if (!mac)
   {
-    Common::GenerateMacAddress(Common::MACConsumer::IOS, mac);
-    SaveMACAddress(mac);
+    mac = Common::GenerateMacAddress(Common::MACConsumer::IOS);
+    SaveMACAddress(mac.value());
     if (!wireless_mac.empty())
     {
       ERROR_LOG(IOS_NET,
                 "The MAC provided (%s) is invalid. We have "
```

Comment:
the error message acts like it's printing the invalid address, but it's actually printing the new generated address

Reply:
This isn't related to the changes being made.